File: rabbitmq/consumer.py
import pika

def main():
    # Connection parameters
    connection_params = pika.ConnectionParameters('localhost')
    
    # Connect to RabbitMQ server
    connection = pika.BlockingConnection(connection_params)
    channel = connection.channel()
    
    # Declare the queue (to ensure it exists)
    queue_name = 'test_queue'
    channel.queue_declare(queue=queue_name, durable=True)

    print(f" [*] Waiting for messages in '{queue_name}'. To exit press CTRL+C")

    # Callback function to process received messages
    def callback(ch, method, properties, body):
        print(f" [x] Received: {body.decode()}")

    # Set up consumer to fetch messages from the queue
    channel.basic_consume(queue=queue_name, on_message_callback=callback, auto_ack=True)

    # Start consuming
    try:
        channel.start_consuming()
    except KeyboardInterrupt:
        print("\n [!] Exiting...")
        channel.stop_consuming()

    # Close the connection
    connection.close()

if __name__ == "__main__":
    main()

File: rabbitmq/consumer.cpp

#include <iostream>
#include <thread>
#include <SimpleAmqpClient/SimpleAmqpClient.h>

int main() {
    try {
        // Connect to RabbitMQ server
        AmqpClient::Channel::ptr_t channel = AmqpClient::Channel::Create("localhost");

        // Queue name
        std::string queue_name = "test_queue";

        // Declare the queue (if it doesn't exist already)
        channel->DeclareQueue(queue_name, false, true, false, false);

        std::cout << " [*] Waiting for messages in queue: " << queue_name << std::endl;

        while (true) {
            // Fetch a single message from the queue
            AmqpClient::Envelope::ptr_t envelope = channel->BasicGet(queue_name, true);

            if (envelope) {
                // Process the received message
                std::string message = envelope->Message()->Body();
                std::cout << " [x] Received: " << message << std::endl;
            } else {
                // No message available
                std::cout << " [*] No messages in the queue. Retrying..." << std::endl;
            }

            // Sleep briefly to avoid busy looping
            std::this_thread::sleep_for(std::chrono::seconds(1));
        }
    } catch (const std::exception& e) {
        // Handle any exceptions
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }

    return 0;

}
